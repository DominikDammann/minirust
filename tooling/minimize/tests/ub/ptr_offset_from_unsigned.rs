#![allow(internal_features)]
#![feature(ptr_sub_ptr, core_intrinsics)]

use std::intrinsics;

fn main() {
    let data = [1u8, 2, 3, 4, 5];
    unsafe {
        // Negative offset in `ptr_offset_from_unsigned` is UB.
        intrinsics::ptr_offset_from_unsigned(&data[0], &data[4]);
    }
}
