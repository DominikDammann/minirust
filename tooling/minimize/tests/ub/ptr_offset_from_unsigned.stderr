fatal error: UB: PtrOffsetFrom: negative result with `nonneg` flag set
